volume-widget/volume.lua

```diff
@@ -56,79 +122,41 @@ local function worker(args)
             self.icon.image = path
         end
     }
-
-    local notification
-    local volume_icon_name="audio-volume-high-symbolic"
-
-    local function get_notification_text(txt)
-        local mute = string.match(txt, "%[(o%a%a?)%]")
-        local volume = string.match(txt, "(%d?%d?%d)%%")
-        volume = tonumber(string.format("% 3d", volume))
-        if mute == "off" then
-            return volume.."% <span color=\"red\"><b>Mute</b></span>"
-        else
-            return volume .."%"
-        end
+--}}}
+--{{{ Spawn functions
+    function volume:_cmd(cmd)
+        notif("")
```

Comment:
What is it for? Should it be here?

Reply:
This will pop up the notification.
Then the async call to update_graphic will change the content of the notification with correct icon and level.
I didn't find a better way to do it with the right value or in the correct scope.
Perhaps I miss something trivial to have the notification correctly popped up from the update_graphic callback but I never find it.